account_invoice_change_currency/models/account_change_currency.py

```diff
@@ -0,0 +1,311 @@
+                rate_value.new_value_float,
+            )
+        if (
+            len(last_values) == 1
+            and last_values.mail_message_id.subtype_id == subtype_create_id
+        ):
+            return (self.currency_id.browse(last_values.new_value_integer), None)
+        return self.currency_id.browse(None), None
+
+    def _toggle_forced_rate(self):
+        self.ensure_one()
+        state = self.get_force_rate_state()
+        if not state:
+            self._track_force_rate()
+        else:
+            force = not state.new_value_integer
+            self._track_force_rate(force)
+
+    def get_force_rate_state(self):
+        self.ensure_one()
+        subtype_id = self.env.ref("account_invoice_change_currency.mt_force_rate")
+        ir_model_fields = self.env["ir.model.fields"]
+        ir_model_fields_id = (
+            ir_model_fields.sudo()
+            .search([("name", "=", "to_check"), ("model", "=", self._name)])
+            .id
+        )
+        domain = [
+            ("mail_message_id", "in", self.message_ids.ids),
+            ("mail_message_id.subtype_id", "=", subtype_id.id),
+            ("field", "=", ir_model_fields_id),
+        ]
+        last_value = (
+            self.env["mail.tracking.value"]
+            .sudo()
+            .search(domain, limit=1, order="write_date desc, id desc")
+        )
+        return last_value
+
+    def _track_force_rate(self, force=True):
+        track = self.env["mail.tracking.value"]
+        force_rate_description = self.fields_get(["to_check"])["to_check"]
+        force_rate_description.update({"string": "Force Rate"})
+        tracking_value_ids = [
+            [
+                0,
+                0,
+                track.create_tracking_values(
+                    None,
+                    force,
+                    "to_check",
+                    force_rate_description,
+                    100,
+                    self._name,
+                ),
+            ],
+            [
+                0,
+                0,
+                track.create_tracking_values(
+                    self.currency_id,
+                    self.currency_id,
+                    "currency_id",
+                    self.fields_get(["currency_id"])["currency_id"],
+                    100,
+                    self._name,
+                ),
+            ],
+            [
+                0,
+                0,
+                track.create_tracking_values(
+                    self.custom_rate,
+                    self.custom_rate,
+                    "rate",
+                    self.fields_get(["custom_rate"])["custom_rate"],
+                    100,
+                    "res.currency",
+                ),
+            ],
+        ]
+        self.message_post(
+            subtype_id=self.env.ref("account_invoice_change_currency.mt_force_rate").id,
+            tracking_value_ids=tracking_value_ids,
+        )
+
+    def track_created_invoice(self):
```

Comment:
Why is this required?

Reply:
The bug I commented once was that the first currency change did not detect the last value when the invoice was created. In Odoo 12 this worked fine, but in Odoo 13 and Odoo 14 it didn't detect the currency when the invoice was created. This function adds this functionality back.